src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -647,6 +642,28 @@ async def test_TC_TSTAT_4_2(self):
         self.step("18")
         if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
 
+            # Generate list of tuples of scenarios and number of remaining presets per scenario allowed
+            presetScenarioHeadroom = list((presetType.presetScenario,
+                                           presetType.numberOfPresets - presetScenarioCounts.get(presetType.presetScenario, 0)) for presetType in presetTypes)
+
+            if len(presetScenarioHeadroom) > 0:
+                # Find the preset scenario with the smallest number of remaining allowed presets
+                presetScenarioHeadroom = sorted(presetScenarioHeadroom, key=lambda diff: diff[1])
```

Comment:
how about a `@dataclass` for these instead of a tuple? figuring out what the indices are is harder to read and maintain.

Reply:
Switched to a namedtuple, as it's only used in this one place.